### PR TITLE
Fix collection fields retreival bug when Views are present in MONGO DB

### PR DESCRIPTION
**Pull request by jodevsa**
Hi,

Retrieval of  collection fields will fail  when a collection of type View is present. This is because $natural sort is being used for all collection types although isn't supported in Views ([link](https://docs.mongodb.com/manual/reference/operator/meta/natural/))

This commit will add functionality to detect  Views and handle them accordingly (first 2 documents are considered the sample)

Thanks.

### CODE REVIEW

File: redash/query_runner/mongodb.py [hunk 1]
```diff
@@ -174,6 +174,13 @@ def _merge_property_names(self, columns, document):
               if property not in columns:
                   columns.append(property)
 
+    def _is_collection_a_view(self, db, collection_name):
+        try:
+            db.command('collstats', collection_name)
+            return True
```
Comment (arikfr) [inline redash/query_runner/mongodb.py]: When collstats succeeded doesn't it mean this is a Collection rather than a View?

Comment (jodevsa) [reply redash/query_runner/mongodb.py]: Your absolutely right, forgot to switch them before committing

File: redash/query_runner/mongodb.py [hunk 2]
```diff
@@ -184,19 +191,20 @@ def _get_collection_fields(self, db, collection_name):
...
-        last_document = None
-
-        for d in db[collection_name].find().sort([("$natural", 1)]).limit(1):
-            first_document = d
-
-        for d in db[collection_name].find().sort([("$natural", -1)]).limit(1):
-            last_document = d
+        collection_is_a_view = self._is_collection_a_view(db, collection_name)
+        documents_sample = []
+        if(collection_is_a_view):
```
Comment (arikfr) [inline redash/query_runner/mongodb.py]: You don't need parentheses around the `if` statement:

```suggestion
        if collection_is_a_view:
```

Comment (jodevsa) [reply redash/query_runner/mongodb.py]: Cool!

**Comment by jodevsa** [comment]
@akariv  Not sure why the build failed this time, it's working from my side.

**Comment by arikfr** [comment]
I'm not sure either. I triggered another execution.

**Comment by arikfr** [comment]
Then I'll trigger another one. 🤷‍♂️

**Comment by jodevsa** [comment]
finally :1st_place_medal:

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**